build/parseSpec.c

```diff
@@ -246,8 +246,8 @@ static int expandMacrosInSpecBuf(rpmSpec spec, int strip)
     if ((condition) && (!condition->withArgs)) {
 	const char *s = lbuf + condition->textLen;
 	SKIPSPACE(s);
-	if (s[0])
-	    rpmlog(RPMLOG_WARNING,
+	if (s[0] && s[0] != '#')
+	    rpmlog(RPMLOG_ERR,
```

Comment:
An error needs to also abort the parse.

Reply:
...and I think we should actually test it does that 

/me ducks

Reply:
No idea why I thought RPMLOG_ERR looked like it already did that...

Reply:
Merged, but the thing is: I have to be able to trust that anything submitted as PR has *at least* been tested manually. Otherwise it's wasting people's time.